File: online-bookstore-server/microservice-bookstore-common/src/main/java/com/onlinebookstore/entity/bookserver/Book.java
package com.onlinebookstore.entity.bookserver;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.onlinebookstore.util.bookutil.BookConstantPool;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * @author rkc
 * @date 2020/9/17 15:50
 * @version 1.0
 */
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Book implements Serializable {

    /**
     * 自增长id
     */
    private Integer id;

    /**
     * 图书名称
     */
    @JsonProperty("book_name")
    private String bookName;

    /**
     * isbn
     */
    private String isbn;

    /**
     * 作者
     */
    private String author;

    /**
     * 出版商
     */
    private String publisher;

    /**
     * 价格
     */
    private Integer price;

    /**
     * 上架时间
     */
    @JsonProperty("create_time")
    @JsonFormat(pattern = BookConstantPool.TIME_FORMAT, timezone = BookConstantPool.TIMEZONE)
    private Date createTime;

    /**
     * 类型id
     */
    @JsonProperty("t_id")
    private Integer typeId;

    /**
     * 主封面
     */
    @JsonProperty("main_cover")
    private String mainCover;

    /**
     * 描述信息
     */
    private String description;

    /**
     * 图书资源列表（图片）
     */
    private List<BookResource> bookResources;

    /**
     * 图书库存信息
     */
    private BookStorage bookStorage;
}

File: online-bookstore-android/app/src/main/java/com/rkc/onlinebookstore/model/book/Book.kt
package com.rkc.onlinebookstore.model.book

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*
import kotlin.collections.ArrayList

/**
 * @author rkc
 * @date 2020/11/11 17:42
 * @version 1.0
 */
@Parcelize
data class Book (
    @SerializedName("id") val id: Int = -1,
    @SerializedName("book_name") val bookName: String = "",
    @SerializedName("isbn") val isbn: String = "",
    @SerializedName("author") val author: String = "",
    @SerializedName("publisher") val publisher: String = "",
    @SerializedName("price") val price: Int = -1,
    @SerializedName("create_time") val createTime: Date = Date(),
    @SerializedName("t_id") val typeId: Int = -1,
    @SerializedName("main_cover") val mainCover: String = "",
    @SerializedName("description") val description: String = "",
    @SerializedName("bookResources") var bookResources: ArrayList<BookResource> = arrayListOf(),
    @SerializedName("bookStorage") var bookStorage: BookStorage = BookStorage()
) : Parcelable